lib/iris/_merge.py

```diff
@@ -1263,9 +1285,15 @@ def register(self, cube, error_on_mismatch=False):
             this :class:`ProtoCube`.
 
         """
-        match = self._cube_signature.match(self._build_signature(cube),
-                                           error_on_mismatch)
+        signature = self._cube_signature
+        other = self._build_signature(cube)
+        match = signature.match(other, error_on_mismatch)
         if match:
+            if signature.fill_value is None and other.fill_value is not None:
+                # Perform proto-cube fill-value promotion.
+                fv = other.fill_value
+                self._cube_signature = self._build_signature(self._source,
+                                                             fill_value=fv)
```

Comment:
@dkillick This is where `fill_value` promotion happens. We promote the proto-cube with a `fill_value` of `None` to match the non-`None` other `fill_value`.